examples/producer-consumer/README.md

```diff
@@ -23,19 +104,140 @@ Linking ./producer-consumer
 
 ## How to Run
 
-Once `producer-consumer` has been compiled, in the same directory as this README file run `./producer-consumer`. You should a set of messages from: Main, Buffer, Producer, and Consumer actors.
+Once `producer-consumer` has been compiled, in the same directory as this README file run `./producer-consumer`. You should a see of a log like:
 
 ```console
 $ ./producer-consumer
-**Main** Finished.
-**Buffer** Permission_to_produce
-**Buffer** Permission_to_produce: Calling producer to produce
-...
-**Buffer** Store_product
-**Buffer** Store_product: Calling consumer to consume
-**Consumer** Consuming product 2
+test: fifo/basic apply() called.
+1 test started, 0 complete: fifo/basic started
+fifo: created with capacity: 2 and size: 2
+consumer: started. _next = 0
+producer: created. will produce 3
+fifo: consumerRequestsNext: nothing for consumer, add them to _waitQcons
+producer: started. _next = 0
+fifo: requestToProduce(next=0) allows producer to produce id = 0 since _buf.size = 2 and _promised = 0 together are < _cap == 2
+producer: has produced 0
+fifo: dispatch() is taking consumer off _waitQcons to provide them next = 0
+fifo: requestToProduce(next=1) allows producer to produce id = 1 since _buf.size = 2 and _promised = 0 together are < _cap == 2
+consumer: has consumed 0
+producer: has produced 1
+consumer: about to ask for _next = 1
+fifo: requestToProduce(next=2) allows producer to produce id = 2 since _buf.size = 2 and _promised = 0 together are < _cap == 2
+fifo: consumerRequestsNext() has _buf.size = 2
+fifo: consumerRequestsNext about to provide = 1 ; now _buf.size = 2
+producer: has produced 2
+consumer: has consumed 1
+consumer: about to ask for _next = 2
+fifo: consumerRequestsNext() has _buf.size = 2
+fifo: consumerRequestsNext about to provide = 2 ; now _buf.size = 2
+consumer: has consumed 2
+elapsed nanosec = 186000
+1 test started, 1 complete: fifo/basic complete
+---- Passed: fifo/basic
+----
+---- 1 test ran.
+---- Passed: 1
 ```
 
 ## Program Modifications
 
-Rather than modifying this program, pay attention to how the use of behaviors means this program is non-blocking and does not use locks/semaphores.
+# basic
+
+1. Read the assertions. Do they assert that the
+consumer never consumes the same product twice?
+If not, add an assertion to this effect.
+
+2. Verify that the ordering invariant is maintained
+in this code. Can we assert this in other places?
+
+In other words, assert that the reader consumes products
+in the same sequential order that the producer produced them.
+
+(Formally, the consumption and production sequences should map one-to-one 
+to the natural numbers.)
+
+3. Try adding more than one consumer. How do the invariants change?
+
+4. Try adding more than one producer. How do the invariants change?
```

Comment:
for proper markdown editor handling, there should be no spaces between the the list items.